examples/dev-example/src/node/model-server-client.ts

```diff
@@ -8,46 +8,44 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR MIT
  *******************************************************************************/
-import { AnyObject, MessageDataMapper, MessageType, SubscriptionOptions } from '@eclipse-emfcloud/modelserver-client';
-import { TheiaBackendModelServerClient } from '@eclipse-emfcloud/modelserver-theia/lib/node';
+import {
+    AnyObject,
+    MessageDataMapper,
+    MessageType,
+    SubscriptionListener,
+    SubscriptionOptionsV2
+} from '@eclipse-emfcloud/modelserver-client';
+import { TheiaBackendModelServerClientV2 } from '@eclipse-emfcloud/modelserver-theia/lib/node';
 import { injectable } from '@theia/core/shared/inversify';
+import URI from 'urijs';
 
 import { DevModelServerClient } from '../common/dev-model-server-client';
 
 @injectable()
-export class CustomDevModelServerClient extends TheiaBackendModelServerClient implements DevModelServerClient {
+export class CustomDevModelServerClient extends TheiaBackendModelServerClientV2 implements DevModelServerClient {
     private intervalId: NodeJS.Timeout;
 
-    subscribe(modeluri: string, options: SubscriptionOptions = {}): void {
+    subscribe(modeluri: URI, listener?: SubscriptionListener, options: SubscriptionOptionsV2 = {}): SubscriptionListener {
```

Comment:
We had multiple examples where the options were set but the listener was undefined. Why use subscription options if no listener is set anyway? This happened in `examples/dev-example/src/browser/api-test-menu.ts`

Reply:
Previously with `v1`, the listener only was a member of the `SubscriptionOptions`. With `v2` it should be a separate argument (for compatibility reasons, the options member will be used as the listener internally then).

> Why use subscription options if no listener is set anyway?

Yes I agree that it might look a bit off, but in the case of the `api-test-menu` example we make use of a `ModelServerSubscriptionClient` that is initalized once and shows notifications received by opened subscriptions.

Perhaps we could re-work the example application a bit (in a separate ticket) and align the listener usage and also show the request (data like command or patch).

Reply:
Could you please open a follow up here as suggested by you?

Reply:
Follow-up ticket: #124